dist/web3-light.js

```diff
@@ -3327,6 +3348,145 @@ module.exports = ICAP;
     You should have received a copy of the GNU Lesser General Public License
     along with ethereum.js.  If not, see <http://www.gnu.org/licenses/>.
 */
+/** @file ipcprovider.js
+ * @authors:
+ *   Fabian Vogelsteller <fabian@ethdev.com>
+ * @date 2015
+ */
+
+"use strict";
+
+var utils = require('../utils/utils');
+var errors = require('./errors');
+
+
+var IpcProvider = function (path, net) {
+    var _this = this;
+    this.responseCallbacks = {};
+    this.path = path;
+    
+    net = net || require('net');
+
+
+    try {
+        this.connection = net.connect({path: this.path});
+    } catch(error) {
+        throw errors.InvalidConnection(path);
+    }
+
+    // this.connection.on('error', function(e){
+    //     throw errors.InvalidConnection(path);
+    // }); 
+
+
+
+    // LISTEN FOR CONNECTION RESPONSES
+    this.connection.on('data', function(result) {
+        result = result.toString();
+
+        try {
+            result = JSON.parse(result);
+
+        } catch(e) {
+            throw errors.InvalidResponse(result);                
+        }
+
+        var id;
+
+        // get the id which matches the returned id
+        if(utils.isArray(result)) {
+            result.forEach(function(load){
+                if(_this.responseCallbacks[load.id])
+                    id = load.id;
```

Comment:
The result can be a batch array and could technically come in  a different order than the send batch. Therefore iterating over the batch results until one items matches the noted ID, is more secure.